### DISCUSSION TITLE
timeout not work in asyncClient

### DISCUSSION THREAD

**Discussion by lyg4795** [encode/httpx#2142] [Potential Issue]
I create a asyncClient  by  'async with httpx.AsyncClient(proxies=proxies,timeout=2.0)as client:'
then I send a request
            print("get cookie res start", f'xid{xid}')
            cookieres = await client.get(url, headers=headers)
            print("get cookie res ",f'xid{xid}',t-time.time())
sometimes the requests will cost more than 2s
            get cookie res  xid0 -4.54197096824646
and sometimes I have to wait several minutes and get an error like httpx.readError

**Comment by AdrianLopezGue** [comment]
Any update with this discussion? timeout does not work with asyncClient and Client 🫤

**Reply by lovelydinosaur** [reply]
Are you able to share an example that we can try?

**Reply by AdrianLopezGue** [reply]
Sure, this example:
``` python
def execute_request(verify: VerifyTypes, request: httpx.Request):
  try:
      with httpx.Client(verify=verify, timeout=1.0) as client:
          webhook_response = client.send(request)
  
          return webhook_response
  
  except httpx.TimeoutException as error:
      raise error
  except Exception as error:
      raise error
```
I have a webhook configured to return a response after 5 seconds and it never raise any exception 🫤

**Reply by lovelydinosaur** [reply]
Ah... in [this bit of the docs](https://www.python-httpx.org/advanced/#request-instances) we should probably recommend preferring the usage of `client.build_request(...)` if you're using `client.send()`.

If you take a look at the source code, you can see that `timeout` is specified *on the request instance*.

https://github.com/encode/httpx/blob/cd1d51d597a3d75cef5b0d9479b5a89d39a51ec0/httpx/_client.py#L808-L821

If you're not using `.build_request()` but instead instantiating the `Request()` directly, then the timeout and other bits of client configuration *won't be automatically applied*.

You need to either use `client.build_request(...)`, or else make sure you're correctly setting `Request(..., extensions={"timeout": ...})`, which is a bit fiddly to get correct.
